Lambdas/GHHooks/GithubEvent.swift

```diff
@@ -0,0 +1,281 @@
+    let avatarURL: String
+    let gravatarID: String
+    let url, htmlURL, followersURL: String
+    let followingURL, gistsURL, starredURL: String
+    let subscriptionsURL, organizationsURL, reposURL: String
+    let eventsURL: String
+    let receivedEventsURL: String
+    let type: String
+    let siteAdmin: Bool
+
+    enum CodingKeys: String, CodingKey {
+        case login, id
+        case nodeID
+        case avatarURL
+        case gravatarID
+        case url
+        case htmlURL
+        case followersURL
+        case followingURL
+        case gistsURL
+        case starredURL
+        case subscriptionsURL
+        case organizationsURL
+        case reposURL
+        case eventsURL
+        case receivedEventsURL
+        case type
+        case siteAdmin
+    }
+}
+
+// MARK: - Permissions
+struct Permissions: Codable {
+    let pull, push, admin: Bool
+}
+
+// MARK: - SecurityAndAnalysis
+struct SecurityAndAnalysis: Codable {
+    let advancedSecurity, secretScanning, secretScanningPushProtection: AdvancedSecurity
+
+    enum CodingKeys: String, CodingKey {
+        case advancedSecurity
+        case secretScanning
+        case secretScanningPushProtection
+    }
+}
+
+// MARK: - AdvancedSecurity
+struct AdvancedSecurity: Codable {
+    let status: String
+}
+
+// MARK: - User
+struct User: Codable {
+    let login: String
+    let id: Int
+    let nodeID: String
+    let avatarURL: String
+    let gravatarID: String
+    let url, htmlURL, followersURL: String
+    let followingURL, gistsURL, starredURL: String
+    let subscriptionsURL, organizationsURL, reposURL: String
+    let eventsURL: String
+    let receivedEventsURL: String
+    let type: String
+    let siteAdmin: Bool
+    let name, company: String
+    let blog: String
+    let location, email: String
+    let hireable: Bool
+    let bio, twitterUsername: String
+    let publicRepos, publicGists, followers, following: Int
+    let createdAt, updatedAt: Date
+
+    enum CodingKeys: String, CodingKey {
+        case login, id
+        case nodeID
+        case avatarURL
+        case gravatarID
+        case url
+        case htmlURL
+        case followersURL
+        case followingURL
+        case gistsURL
+        case starredURL
+        case subscriptionsURL
+        case organizationsURL
+        case reposURL
+        case eventsURL
+        case receivedEventsURL
+        case type
+        case siteAdmin
+        case name, company, blog, location, email, hireable, bio
+        case twitterUsername
+        case publicRepos
+        case publicGists
+        case followers, following
+        case createdAt
+        case updatedAt
+    }
+}
+
+//MARK: - DereferenceBox
+final class DereferenceBox<C: Codable>: Codable {
+    var value: C
+
+    init(from decoder: any Decoder) throws {
+        self.value = try .init(from: decoder)
+    }
+
+    func encode(to encoder: any Encoder) throws {
+        try self.value.encode(to: encoder)
+    }
```

Comment:
This is an _**extremely**_ fine distinction that is completely lost on most uses (and users) of `Codable`, since it only makes a difference in very atypical cases, but those cases do actually come up now and then. The correct way to do this is:
```suggestion
    init(from decoder: any Decoder) throws {
        let container = try decoder.singleValueContainer()
        self.value = try container.decode(C.self)
    }

    func encode(to encoder: any Encoder) throws {
        var container = encoder.singleValueContainer()
        try container.encode(self.value)
    }
```

Reply:
I actually had noticed the decoding difference, but I couldn't think of it when I was deciding which version to use, so just went with the more-consice version.

Still don't remember what it was.